tests/test_triton_blocksparse.py

```diff
@@ -123,17 +120,18 @@ def test_softmax(BLOCK, WIDTH, DTYPE):
     ry = block_sparsify_tensor(ry, layout, BLOCK)
 
     # compare
-    assert_almost_equal(ry, ty)
+    triton.testing.assert_almost_equal(ry, ty)
 
 
 @pytest.mark.skipif(not _triton_available, reason="Triton requires a recent CUDA gpu")
 @pytest.mark.parametrize("block", [32, 43, 128])  # 16, 32,
+@pytest.mark.parametrize("dtype", [torch.float16])
```

Comment:
parity is not respected with bfloat16, but this amounts to testing triton's blocksparse really so I'm not sure that this is xformers' job really. Lack of parity with pytorch could be explained by different accumulator choices (tf32, float32, ..) and/or normalization in different orders, I tried the obvious suspects to no avail. In the end I thought that it was fair to delegate that to triton, this xformer layer is just a wrapper